### PR TITLE
feat(discover-snql): Add support for filtering on message

### PR DESCRIPTION
**Pull request by Zylphrex**
This adds support for filter on the message field.

### CODE REVIEW

File: src/sentry/search/events/filter.py [hunk 1]
```diff
@@ -1261,6 +1262,44 @@ def _environment_filter_converter(self, search_filter: SearchFilter) -> Optional
...
+        else:
+            # https://clickhouse.yandex/docs/en/query_language/functions/string_search_functions/#position-haystack-needle
+            # positionCaseInsensitive returns 0 if not found and an index of 1 or more if found
+            # so we should flip the operator here
+            operator = Op.NEQ if search_filter.operator in EQUALITY_OPERATORS else Op.EQ
+            if search_filter.is_in_filter:
+                return Condition(
+                    Function(
+                        "multiSearchFirstPositionCaseInsensitive",
+                        [self.column("message"), Function("array", value)],
```
Comment (shruthilayaj) [inline src/sentry/search/events/filter.py:1289]: If `value` is an array here, you don't need to cast it as array, you should be ok to just do:
```suggestion
                        [self.column("message"), value],
```

File: src/sentry/search/events/filter.py [hunk 2]
```diff
@@ -1261,6 +1262,44 @@ def _environment_filter_converter(self, search_filter: SearchFilter) -> Optional
...
+                    0,
+                )
+
+            # make message search case insensitive
+            return Condition(
+                Function("positionCaseInsensitive", [self.column("message"), value]),
+                operator,
+                0,
+            )
+        return None
```
Comment (shruthilayaj) [inline src/sentry/search/events/filter.py:1301]: Should we remove this to just implicitly return None? The Optional return type implies that the function can return None, so we don't need to explicitly add it here

**Review by shruthilayaj — APPROVED** [review]
Minor comments, LGTM

File: src/sentry/search/events/filter.py (hunk 1 above)
Comment (Zylphrex) [reply src/sentry/search/events/filter.py:1289]: Nice, TIL

**Pull request merged**